### PR TITLE
Session auth

### PR DESCRIPTION
**Pull request by rnv19**

### CODE REVIEW

File: Project1/project1/.env [hunk 1]
```diff
@@ -0,0 +1,3 @@
+FLASK_APP=application.py
+DATABASE_URL=postgres://nesrqsgxfpkmhs:837e8074d7ce3afe50e5e7ced003d7e2a9440c3a92647793a2225179fe0d847e@ec2-54-147-209-121.compute-1.amazonaws.com:5432/desi636rresb9m
```
Comment (manasa133) [inline Project1/project1/.env:2]: remove the url, it needs to be given from cmd for security reasons

File: Project1/project1/models.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from flask_sqlalchemy import SQLAlchemy
+
+db = SQLAlchemy()
+
+class Users(db.Model):
+    __tablename__ = "USERS"
```
Comment (manasa133) [inline Project1/project1/models.py:6]: for authentication username and passwprd will be enough to store in db

Comment (manasa133) [reply Project1/project1/models.py:6]: time stamp is missing

File: Project1/project1/application.py [hunk 3]
```diff
@@ -33,31 +33,60 @@ def main():
...
+            return render_template("registration.html")
+        
+        elif (request.form["button"] == "submit"):
+            name = request.form.get("name")
+            username = request.form.get("username")
+            password = request.form.get("password")
+            birthday = request.form.get("birthday")
+            address = request.form.get("address")
+            print(name, username, password, birthday, address)
+            user = Users(name=name, username=username, password=password, birthday=birthday, address=address)
```
Comment (manasa133) [inline Project1/project1/application.py]: i

File: Project1/project1/application.py [hunk 4]
```diff
@@ -33,31 +33,60 @@ def main():
...
+            try:
+                db.session.add(user)
+                db.session.commit()
+                return (f"User {name} successfully added to database...!")
+            except Exception as e:
```
Comment (manasa133) [inline Project1/project1/application.py]: where is this exception returned? can the user see

File: Project1/project1/application.py [hunk 5]
```diff
@@ -33,31 +33,60 @@ def main():
...
+                        session["username"] = res.username
+                        print("after login", session)
+                        return render_template("user.html", username = username)
+            except Exception as e:
+                print(e)
+                return render_template("login.html", alert=True)
 
 @app.route("/admin", methods=["GET"])
 def admin():
     res = Users.query.all()
```
Comment (manasa133) [inline Project1/project1/application.py]: need to display recently registerd user first

File: Project1/project1/import.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+app = Flask(__name__)
+app.config["SQLALCHEMY_DATABASE_URI"] = os.getenv("DATABASE_URL")
+app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
+db = SQLAlchemy(app)
+# db.init_app(app)
+
+from models import *
+
+def main():
+    db.create_all()
```
Comment (manasa133) [inline Project1/project1/import.py]: this is redundant as its already done in application.py

File: Project1/project1/.env (hunk 1 above)
Comment (rnv19) [reply Project1/project1/.env:2]: Data removed

File: Project1/project1/models.py (hunk 2 above)
Comment (rnv19) [reply Project1/project1/models.py:6]: removed birthday and address column from database, added created_at timestamp column

File: Project1/project1/application.py (hunk 4 above)
Comment (rnv19) [reply Project1/project1/application.py]: This exception is returned to user display, yes user can see the error message

File: Project1/project1/application.py (hunk 5 above)
Comment (rnv19) [reply Project1/project1/application.py]: Necessary changes done

File: Project1/project1/import.py (hunk 6 above)
Comment (rnv19) [reply Project1/project1/import.py]: import.py is a separate python file which i wrote to load the book details into the database as mentioned in LMS. That's why i wronte db.create.all() again in import.py.

Comment (manasa133) [reply Project1/project1/import.py]: your already writing db.createall() for users table also, so make sure there it self the books table is also created. remove import.py or remove this part from application.py

Comment (rnv19) [reply Project1/project1/import.py]: Removed db.createall() from import.py and integrated in application.py

**Pull request merged**